magda-search-api/src/main/scala/au/csiro/data61/magda/search/elasticsearch/ElasticSearchQueryer.scala

```diff
@@ -105,51 +107,33 @@ class ElasticSearchQueryer(indices: Indices = DefaultIndices)(
     clientFuture.flatMap { implicit client =>
       val fullRegionsFutures = inputRegionsList.map(resolveFullRegion)
       val fullRegionsFuture = Future.sequence(fullRegionsFutures)
-      val query = buildQueryWithAggregations(
-        inputQuery,
-        start,
-        limit,
-        MatchAll,
-        requestedFacetSize)
-
-      Future.sequence(
-        Seq(
-          fullRegionsFuture,
-          client.execute(query).flatMap {
-            case Right(results) =>
-              if (results.result.totalHits > 0)
-                Future.successful((results.result, MatchAll))
-              else
-                client
-                  .execute(
-                    buildQueryWithAggregations(
-                      inputQuery,
-                      start,
-                      limit,
-                      MatchPart,
-                      requestedFacetSize))
-                  .map {
-                    case Right(results) => (results.result, MatchPart)
-                    case Left(IllegalArgumentException(e)) => throw e
-                    case Left(ESGenericException(e)) => throw e
-                  }
-            case Left(IllegalArgumentException(e)) => throw e
-            case Left(ESGenericException(e)) => throw e
-          }))
-    } map {
-      case Seq(fullRegions: List[Option[Region]],
-        (response: SearchResponse, strategy: SearchStrategy)) =>
-        val newQueryRegions =
-          inputRegionsList
-            .zip(fullRegions)
-            .filter(_._2.isDefined)
-            .map(_._2.get)
-            .map(Specified.apply)
-            .toSet ++ inputQuery.regions.filter(_.isEmpty)
-
-        val outputQuery = inputQuery.copy(regions = newQueryRegions)
-
-        buildSearchResult(outputQuery, response, strategy, requestedFacetSize)
+      augmentWithBoostRegions(inputQuery).flatMap { queryWithBoostRegions =>
+        val query = buildQueryWithAggregations(queryWithBoostRegions, start, limit, MatchAll, requestedFacetSize)
+        Future.sequence(Seq(fullRegionsFuture, client.execute(query).flatMap {
+          case Right(results) =>
+            if (results.result.totalHits > 0)
+              Future.successful((results.result, MatchAll))
+            else
+              client.execute(buildQueryWithAggregations(queryWithBoostRegions, start, limit, MatchPart, requestedFacetSize)).map {
+                case Right(results) => (results.result, MatchPart)
+                case Left(IllegalArgumentException(e)) => throw e
+                case Left(ESGenericException(e)) => throw e
+              }
+          case Left(IllegalArgumentException(e)) => throw e
+          case Left(ESGenericException(e)) => throw e
+        })).map {
+          case Seq(fullRegions: List[Option[Region]], (response: SearchResponse, strategy: SearchStrategy)) =>
+            val newQueryRegions =
```

Comment:
Can you put a comment in saying what's happening here? I don't quite understand it.